pkg/vault/contracts/Vault.sol

```diff
@@ -391,6 +397,16 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
                 revert SwapLimit(amountOutRaw, params.limitRaw);
             }
         } else {
+            // Set locals.swapFeeAmountScaled18 based on the amountCalculated.
+            if (state.swapFeePercentage > 0) {
+                // To ensure symmetry with EXACT_IN, the swap fee used by ExactOut is
+                // `fee%/(100% - fee%) * amountCalculated`. Add it to the calculated amountIn. Round up to avoid losses
+                // during precision loss.
+                locals.swapFeeAmountScaled18 =
+                    (amountCalculatedScaled18 * state.swapFeePercentage) /
+                    (FixedPoint.ONE - state.swapFeePercentage);
```

Comment:
```suggestion
                locals.swapFeeAmountScaled18 = amountCalculatedScaled18.mulDivUp(state.swapFeePercentage, state.swapFeePercentage.complement());
```

As the comment indicates, we should probably round up.
`mulDivUp` does just that: raw multiplication, then div up.

If `swapFeePercentage` is ONE or greater, the complement will be 0 and `mulDivUp` will revert. We shouldn't need that protection, but it should be ok.

Reply:
It fits with just this change!

Reply:
It fits with room to spare and all tests pass if you this and remove the if's. (swapFeeAmounts initialize to 0 anyway, so if it's ever 0, it will just multiply 0 by 0.)

```
locals.swapFeeAmountScaled18 =
    (amountCalculatedScaled18 * state.swapFeePercentage) /
    state.swapFeePercentage.complement();
```
The equivalent `mulDivUp` is failing tests, though?

Reply:
```
(amountCalculatedScaled18 * state.swapFeePercentage) / (FixedPoint.ONE - state.swapFeePercentage);
```

This is equivalent to rounding down, not up.

Reply:
Tests also round down, which is why they're failing, then.

Reply:
Ok, fixed the code and tests.